internal/reconcile/helmchart_template.go

```diff
@@ -95,6 +95,20 @@ func (r *HelmChartTemplate) Reconcile(ctx context.Context, req *Request) error {
 		}
 	}
 
+	if mustCleanDeployedChart(obj) {
```

Comment:
I would simply let the existence of a reference take precedence over the template existing.

Reply:
My assumption is that if a user switch to a `chartRef` and wants to use an existing `ociRepo` it is because he wants to reduce the number of existing `helmChart` resources. So if we don't clean up during the reconciliation, the user will likely perform the clean up by other means. That would leave a `status` that does not reflect the actual state.